qiita_db/artifact.py

```diff
@@ -1550,6 +1551,27 @@ def being_deleted_by(self):
             res = qdb.sql_connection.TRN.execute_fetchindex()
         return qdb.processing_job.ProcessingJob(res[0][0]) if res else None
 
+    @property
+    def has_human(self):
+        has_human = False
+        if self.artifact_type == 'per_sample_FASTQ':
+            st = self.study.sample_template
+            if 'env_package' in st.categories:
```

Comment:
I worry about whether this is stringent enough. Users who upload data, or data we import from EBI, is not assured to have the `env_package` variable. If the study lacks this variable then we may accidentally be exposing human associated data. Are there any human associated shotgun samples in Qiita right now that this doesn't catch?

One thing we could do is test multiple variables, such as also including `host_taxid`, `host_scientific_name`, and `host_common_name`. However, I think it would be helpful to take a data driven perspective here and make sure what we choose is able to capture all human shotgun data we currently index

Reply:
Good question! As background, around 4 years ago we introduced the [idea of restrictions to change artifacts](https://github.com/qiita-spots/qiita/blob/master/qiita_db/support_files/patches/74.sql#L25) from sandbox to private/public so it means that all studies that have changed level since that date should have env_package. Now that doesn't mean that this is the case for older studies; however,  all Metagenomic data has been added in more recent years.

Anyway, to give do this more data oriented, we currently have: Counter({'public': 695, 'private': 153, 'sandbox': 3254}) and checking Study status and the existence of env_package & host_taxid, we have:
Counter({'public host_taxid': 38,
         'public env_package': 638,
         'private host_taxid': 3,
         'sandbox host_taxid': 150,
         'sandbox env_package': 889,
         'private env_package': 130})

What do you think?

Reply:
Are any of the public studies which lack env_package ones we should be concerned about? Are there other variables those studies have which could be used?

Reply:
IMOO yes but I think we should add that column to the studies vs. trying to use another column. I'll send an email to the qiita.admin's with the studies.